### PR TITLE
Added AnswersStudentGroup table and desc in questions table

### PR DESCRIPTION
**Pull request by harisibrahimkv**

### CODE REVIEW

**Comment by harisibrahimkv** [comment]
@shivangi The `blank=True` parameter is only considered exclusively when Django is trying to render forms with the corresponding model. So don't worry about that parameter when you're thinking from a Database level. `null=True` is what sets `NULL` or `NOT NULL` constraint on the specific db column. So the question to be asked is "Should we allow that field to be `null=True` at all?"

Will leave individual replies to the other comments.

File: apps/schools/models/assessments.py [hunk 1]
```diff
@@ -26,11 +26,11 @@ class Assessment(models.Model):
...
     end_date = models.DateField(max_length=20, default=default_end_date)
-    query = models.CharField(max_length=500, blank=True, null=True)
-    active = models.IntegerField(blank=True, null=True, default=2)
+    query = models.CharField(max_length=500, blank=True, null=True) #Shivangi, is this needed?
+    active = models.IntegerField(blank=True, null=True, default=2) #Shivangi, should this be null or blank?
     double_entry = models.BooleanField('Requires double entry',
             default=True)
-    institutions = models.ManyToManyField(
+    institutions = models.ManyToManyField( #Shivangi, why is this needed?
         'Institution',
```
Comment (harisibrahimkv) [inline apps/schools/models/assessments.py:34]: This is not a DB level field. In the database, we have an `AssessmentInstitutionAssociation` table. However, this `institutions` field on the `Assessment` model helps us to run queries easily using the ORM. For example, to get all the institutions related to one Assessment, we can do something like `assessment_instance.institutions.all()`. So yes, this is needed.

**Comment by harisibrahimkv** [comment]
@shivangi To add to that, some fields will have `null=True` set so that at the time of creating that record, you don't have to pass that value in. Otherwise it will throw a `NOT NULL CONSTRAINT VIOLATED` or so error.

For example, based on a request, you might want to create an institution like:

`inst = Institution.objects.create(name="Xavier's Institute for gifted children")`

Then, based on some logic to be calculated, you might decided whether that Institution should be `active` or not in the next step. So you would do something like:

`inst.active = 1`
`inst.save()`

However, this workflow won't be possible if there is no `null=True` defined on the `active` field.

Now, coming to Tada and the specific comments you've left regarding this, I am not sure if what I explained above is the reason for certain fields to have null=True. However, if they have a `default=something` set, then that field won't be `null` unless someone goes and explicitly sets that field to have a `null` value.

This has a nice explanation of `blank`, `null` and `default`: http://stackoverflow.com/a/4384131

File: apps/schools/models/students.py [hunk 2]
```diff
@@ -41,13 +44,13 @@ class StudentGroup(models.Model):
     section = models.CharField(
         max_length=10,
         choices=Alpha_list,
-        blank=True,
+        blank=True, #Shivangi Check if this should be there!
         null=True,
-        default=''
+        default='' #Shivangi, should there be No Section? Or just null?
```
Comment (harisibrahimkv) [inline apps/schools/models/students.py:49]: If you want to have an empty CharField (read that as "if you want to have an empty string") in the DB, then you should always explicitly use an `empty string` to indicate that. From the Django docs:

```
Avoid using null on string-based fields such as CharField and TextField unless you have an excellent reason. If a string-based field has null=True, that means it has two possible values for “no data”: NULL, and the empty string. In most cases, it’s redundant to have two possible values for “no data;” Django convention is to use the empty string, not NULL.
```

The question however would be whether a section should be allowed to be empty at all.

File: apps/schools/models/students.py [hunk 3]
```diff
@@ -132,9 +135,9 @@ def get_student(self):
 
 
     def get_name(self):
-        return self.child.first_name
+        return self.child.first_name #shivangi, self.first_name?
```
Comment (harisibrahimkv) [inline apps/schools/models/students.py:138]: Yep, that is a bug.

**Comment by harisibrahimkv** [comment]
Some of the `model methods` (functions on Models like the `get_child` one on the `Student` model)  seem to be legacy ones that have been ported over from the old EMS. We can actually scrap most of them once we are done with our first production ready version of Tada. Doesn't hurt anyone to leave the model methods there as it is for now.

**Comment by harisibrahimkv** [comment]
So Brijesh will be taking a look at all the `is this needed?` comments you've left @shivangi.

File: apps/schools/models/assessments.py (hunk 1 above)
Comment (brijeshpoojary) [inline apps/schools/models/assessments.py:34]: I am not sure why ```query```  actually exists in the db. There are no values for it too. Its either null or empty. We can get rid of it i guess.

File: apps/schools/models/assessments.py [hunk 4]
```diff
@@ -162,8 +163,8 @@ class AnswerStudent(models.Model):
             blank=True, null=True)
     last_modified_by = models.ForeignKey(User, blank=True, null=True,
             related_name='last_modified_answer_student')
-    flexi_data = models.CharField(max_length=30, blank=True, null=True)
-    active = models.IntegerField(blank=True, null=True, default=2)
+    flexi_data = models.CharField(max_length=30, blank=True, null=True) #Shivangi, is this needed?
```
Comment (brijeshpoojary) [inline apps/schools/models/assessments.py:166]: This is not really necessary now i guess. This was a feature added to EMS by mahiti and us, which was not really useful.

Comment (harisibrahimkv) [reply apps/schools/models/assessments.py:166]: @brijeshpoojary Do you know what it was being used for?

Comment (brijeshpoojary) [reply apps/schools/models/assessments.py:166]: i actually dont remember the whole purpose of it. @shivangi might be knowing better. some sort of categorising done to a particular assessment. It is a check box to select while creating a new assessment. But we never used it. I dont think we did really implement on this.

File: apps/schools/models/assessments.py (hunk 1 above)
Comment (shivangi) [reply apps/schools/models/assessments.py:34]: Ok

File: apps/schools/models/students.py (hunk 2 above)
Comment (shivangi) [reply apps/schools/models/students.py:49]: We can have a default of 'No Section' , null as false and remove ' ' from the alpa list

File: apps/schools/models/assessments.py (hunk 4 above)
Comment (shivangi) [reply apps/schools/models/assessments.py:166]: It was to enter some different kind of assessment data. I cant remember the logic behind it though. Might be for the data that is to be entered everyday for someone like akshay patra etc.

**Pull request merged**